x-pack/plugins/data_enhanced/server/search/session/check_running_sessions.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  Logger,
+  ElasticsearchClient,
+  SavedObjectsFindResult,
+  SavedObjectsClientContract,
+} from 'kibana/server';
+import { BackgroundSessionStatus, BackgroundSessionSavedObjectAttributes } from '../../../common';
+import { BACKGROUND_SESSION_TYPE } from '../../saved_objects';
+import { getSearchStatus } from './get_search_status';
+import { getSessionStatus } from './get_session_status';
+
+export async function checkRunningSessions(
+  savedObjectsClient: SavedObjectsClientContract,
+  client: ElasticsearchClient,
+  logger: Logger
+): Promise<void> {
+  try {
+    const runningBackgroundSearchesResponse = await savedObjectsClient.find<BackgroundSessionSavedObjectAttributes>(
+      {
+        type: BACKGROUND_SESSION_TYPE,
+        search: BackgroundSessionStatus.IN_PROGRESS.toString(),
+        searchFields: ['status'],
+        namespaces: ['*'],
+      }
+    );
+
+    if (!runningBackgroundSearchesResponse.total) return;
+
+    logger.debug(`Found ${runningBackgroundSearchesResponse.total} running sessios`);
+
+    const updatedSessions = new Array<
+      SavedObjectsFindResult<BackgroundSessionSavedObjectAttributes>
+    >();
+
+    await Promise.all(
+      runningBackgroundSearchesResponse.saved_objects.map(async (session) => {
+        const searchIds = Object.values(session.attributes.idMapping);
+        const searchStatuses = await Promise.all(
+          searchIds.map(async (searchId: string) => {
+            return await getSearchStatus(client, searchId);
```

Comment:
Are there plans to have an async search API that can return multiple statuses for any number of IDs provided? I see the [ES documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/async-search.html#get-async-search) only mentions one form of `GET` for async search which takes one ID only.

Since this loops through every running background session and calls ES with its ID, it could be a lot more optimal. :)

Reply:
@jimczi and I have discussed it before.
It might happen, but still not sure when ;)